### PR TITLE
fix: container-image integration with knative

### LINKED ISSUE #7736: Generated resources for knative do not take quarkus.container-image.group into consideration
**Describe the bug**

From @lordofthejars:

Adding the `kubernetes` extension with the following config:

```
quarkus.kubernetes.deployment-target=knative
quarkus.container-image.group=lordofthejars
```

Results in the following container generated:

```
"revisionTemplate" : {
          "spec" : {
            "container" : {
              "image" : "alex/greeting-knative:1.0-SNAPSHOT",
              "imagePullPolicy" : "IfNotPresent"
            }
          }
        }
```

### PR DESCRIPTION
**Pull request by iocanel**
Fixes: #7736 

The generated resources for knative are lacking container names.
This means that all container specific decorators are broken.
Until, this is addressed in Dekorate itself, this pull request adds the missing name.

The pr also includes integration test for the issue as described in #7736.

### CODE REVIEW

**Comment by iocanel** [comment]
cc @lordofthejars

**Comment by lordofthejars** [comment]
Great, thank you very much. I am going to open a new PR which is not urgent and it will require a deep development in Dekorate, so no worries, not urgent but to have it in mind.

**Comment by gsmet** [comment]
@geoand can you review that one?

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java [hunk 1]
```diff
@@ -374,14 +374,13 @@ private void applyBuildItems(Session session,
...
-        containerImageBuildItem
-                .ifPresent(c -> session.resources().decorate(KNATIVE,
-                        new ApplyContainerImageDecorator(knativeName, c.getImage())));
+        session.resources().decorate(KNATIVE, new AddMissingContainerNameDecorator(knativeName, knativeName));
+
+        containerImageBuildItem.ifPresent(c -> {
+            session.resources().decorate(OPENSHIFT, new ApplyContainerImageDecorator(openshiftName, c.getImage()));
+            session.resources().decorate(KUBERNETES, new ApplyContainerImageDecorator(kubernetesName, c.getImage()));
+            session.resources().decorate(KNATIVE, new ApplyContainerImageDecorator(knativeName, c.getImage()));
+        });
```
Comment (geoand) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java]: Nice improvement :)

**Review by geoand — APPROVED** [review]
Good one, thanks!

**Pull request merged**